sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -693,6 +719,12 @@ public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
 
     @Override
     public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offersToProcess) {
+        if (!apiServerDisabled.get() && (apiServer == null || !apiServer.isStarted())) {
+            LOGGER.info("Waiting for API Server to start ...");
+            declineOffers(driver, Collections.emptyList(), offersToProcess);
+            return;
```

Comment:
There should probably be a check here: exit the scheduler process if the API server is still down after some duration (60s?). The previous code in getResources() looks like it was effectively doing this.